clients/java/signalr/src/test/java/com/microsoft/aspnet/signalr/test/JsonHubProtocolTest.java

```diff
@@ -161,23 +183,89 @@ public void ParseTwoMessages() {
 
         assertEquals("two", invocationMessage2.getTarget());
         assertEquals(null, invocationMessage2.getInvocationId());
-        JsonArray secondMessageResult = (JsonArray) invocationMessage2.getArguments()[0];
-        assertEquals(43, secondMessageResult.getAsInt());
+        int secondMessageResult = (int)invocationMessage2.getArguments()[0];
+        assertEquals(43, secondMessageResult);
     }
 
     @Test
-    public void ParseSingleMessageMutipleArgs() {
+    public void ParseSingleMessageMutipleArgs() throws Exception {
         String stringifiedMessage = "{\"type\":1,\"target\":\"test\",\"arguments\":[42, 24]}\u001E";
-        HubMessage[] messages = jsonHubProtocol.parseMessages(stringifiedMessage);
+        TestBinder binder = new TestBinder(new InvocationMessage("test", new Object[] { 42, 24 }));
+
+        HubMessage[] messages = jsonHubProtocol.parseMessages(stringifiedMessage, binder);
 
         //We know it's only one message
         assertEquals(HubMessageType.INVOCATION, messages[0].getMessageType());
 
         InvocationMessage message = (InvocationMessage)messages[0];
         assertEquals("test", message.getTarget());
         assertEquals(null, message.getInvocationId());
-        JsonArray messageResult = ((JsonArray) message.getArguments()[0]);
-        assertEquals(42, messageResult.get(0).getAsInt());
-        assertEquals(24, messageResult.get(1).getAsInt());
+        int messageResult = (int) message.getArguments()[0];
+        int messageResult2 = (int) message.getArguments()[1];
+        assertEquals(42, messageResult);
+        assertEquals(24, messageResult2);
+    }
+
+    @Test
+    public void ParseMessageWithOutOfOrderProperties() throws Exception {
```

Comment:
Don't make it worse Brennan! Put some camel in there :).

Reply:
Ugh!!!!!